### ISSUE TITLE
Require log messages are constant strings

### ISSUE THREAD

**Issue by omsmith** [Brightspace/D2L.CodeStyle#518]
The messages that are passed to `ILogger`'s logging methods are intended to be constant values. If the string includes details specific to that particular instance of the error, then it is much harder to track error counts, filter out "uninteresting" errors in alerting and other such things. Additional context is meant to be provided through the `Func<T> metadata` parameter.

An analyzer could be made which inspects `IInvocationOperation`s which have a `TargetMethod` matching one of `D2L.LP.Logging.ILogger`'s methods. `SemanticModel`'s `GetConstantValue` can be used to get the value of the `message` argument. If the model is able to return a constant value, all is well, but otherwise should raise a diagnostic.

```csharp
internal sealed class MyClass {

  private static readonly D2L.LP.Logging.ILogger m_log = D2L.LP.Logging.LogManager.GetLogger( ... );

  private static readonly string STATIC_STRING = "static_string";
  private const string CONST_STRING = "const_string";

  void Foo( string myVar ) {
    m_log.Error( "Foo" );
    m_log.Info( CONST_STRING );
    m_log.Warn( CONST_STRING + "Foo" );

    m_log.Error( myVar ); // D2LXXXX: log messages should be constant
    m_log.Error( "Foo" + myVar ); // D2LXXXX: log messages should be constant
    m_log.Error( $"{myVar}" ); // D2LXXXX: log messages should be constant
    m_log.Warn( STATIC_STRING ); // D2LXXXX: log messages should be constant
  }
}
```
